300-quarkus-vertx-webClient/src/test/java/io/quarkus/qe/vertx/webclient/ChuckNorrisResourceTest.java

```diff
@@ -0,0 +1,85 @@
+    public void getChuckJokeAsJSON(){
+
+        stubFor(get(urlEqualTo("/jokes/random"))
+                .willReturn(aResponse()
+                        .withHeader("Accept", "application/json")
+                        .withBody(String.format("{\"categories\":[]," +
+                                "\"created_at\":\"2020-01-05 13:42:19.576875\"," +
+                                "\"icon_url\":\"https://assets.chucknorris.host/img/avatar/chuck-norris.png\"," +
+                                "\"id\":\"%s\"," +
+                                "\"updated_at\":\"2020-01-05 13:42:19.576875\"," +
+                                "\"url\":\"https://api.chucknorris.io/jokes/sC09X1xQQymE4SciIjyV0g\"," +
+                                "\"value\":\"%s\"}", EXPECTED_ID, EXPECTED_VALUE))));
+
+        given()
+                .when()
+                .get("/chuck")
+                .then()
+                .statusCode(200)
+                .body(is("{\"ID\":\"aBanNLDwR-SAz7iMHuCiyw\",\"jokeText\":\"Chuck Norris has already been to mars; that why there's no signs of life\"}"));
+    }
+
+    @Test
+    @DisplayName("Vert.x WebClient [flavor: mutiny] -> Mapped json response by 'as' mutiny method.")
+    public void getChuckJokeByJsonBodyCodec(){
+
+        stubFor(get(urlEqualTo("/jokes/random"))
+                .willReturn(aResponse()
+                        .withHeader("Accept", "application/json")
+                        .withBody(String.format("{\"categories\":[]," +
+                                "\"created_at\":\"2020-01-05 13:42:19.576875\"," +
+                                "\"icon_url\":\"https://assets.chucknorris.host/img/avatar/chuck-norris.png\"," +
+                                "\"id\":\"%s\"," +
+                                "\"updated_at\":\"2020-01-05 13:42:19.576875\"," +
+                                "\"url\":\"https://api.chucknorris.io/jokes/sC09X1xQQymE4SciIjyV0g\"," +
+                                "\"value\":\"%s\"}", EXPECTED_ID, EXPECTED_VALUE))));
+
+        given()
+                .when()
+                .get("/chuck/bodyCodec")
+                .then()
+                .statusCode(200)
+                .body(is(String.format("{\"ID\":\"%s\",\"jokeText\":\"%s\"}", EXPECTED_ID, EXPECTED_VALUE)));
+    }
+
+    @Test
+    @DisplayName("Vert.x WebClient [flavor: mutiny] -> If third party server exceed http client timeout, then throw a timeout exception.")
+    public void getTimeoutWhenResponseItsTooSlow(){
+        final int delay = 3500; // must be greater than vertx.webclient.timeout-sec
+        final int expectedCode = 408;
+
+        stubFor(get(urlEqualTo("/jokes/random"))
+                .willReturn(aResponse()
+                        .withHeader("Accept", "application/json")
+                        .withFixedDelay(delay)));
+
+        given()
+                .when()
+                .get("/chuck/bodyCodec")
+                .then()
+                .statusCode(expectedCode);
```

Comment:
Why not using directly the code here? 
Also, I think it's better to use: Response.Status.REQUEST_TIMEOUT.getStatusCode()